Add test for EventViewSet

jangjichang opened the pull request:
added a test when category field given to EventViewSet.

jangjichang committed:
Add test for EventViewSet

added a test when category field given to EventViewSet.

jangjichang commented on event/tests/test_views.py:
```diff
@@ -25,3 +25,96 @@ def test_event_list(self):
         pass
+
+
+class TestDevEventSearchByCategory(TestCase):
```
주어진 category와 동일한 event를 반환해주는 API에 대한 테스트를 추가했어요.

다음부터는 테스트를 먼저 작성하고 view를 작성할건데요. 그 전에 이미 구현한 category로 검색하는 기능에 대해 테스트를 추가해 봤어요.

`TestDevEventSearchByCategory` 클래스에 5가지의 다른 상황에 대해 테스트를 하는데 각 함수마다 레코드를 생성하면서 테스트하는게 최선인지 의문이에요.

setup에서 레코드를 생성하면 다른 테스트가 진행될 때 데이터가 남아있지 않고 삭제되서 setup에서 레코드를 생성하는 코드를 작성하는건 좋지 않은 것 같고요.

지금은 `TestDevEventSearchByCategory` 클래스에 함수 5개를 만들었는데 차라리 함수 5개를 각각의 클래스로 만들고 setup에서 데이터를 생성하는게 좋을지 고민하고 있어요!

![image](https://user-images.githubusercontent.com/29423339/70344784-e0487b00-189d-11ea-9b6d-0ef48580c7e5.png)

geusan replied:
나 회사에서 장고 테스트 적용할 때, json 구조로 테스트 적용했었어 jsnoschema 라는걸로 응답 구조를 지정하고, 알맞는 형식을 주는지 테스트 하는거야.

django.test.client 인가 client 쓰면 요청에 대한 응답 테스트가 가능해서 편했어.  데이터가 없다면 추가로 넣어야 겠지만, 이미 데이터가 있는 상황이라면 굳이 데이터를 추가할 필요가 없어
https://python-jsonschema.readthedocs.io/en/stable/

Eunhyang replied:
음 `setup`에서 만든 데이터는 테스트 함수 하나마다 새로 생성될텐데?
그리고 `self.category = baker.make('event.Category')`에서 보통 `self`로 쓰는건 클래스단위로 쓰기위해선데 해당함수에서만 사용하려면  클래스 변수를 사용할 필요는 없는것같아!

Eunhyang replied:
흠 에러나네.. 이부분 내가 한번더 찾아볼게!

Eunhyang replied:
지창아 `setup` 아니고 `setUp`이야! 그래서 `setup`함수가 안돌고있었어ㅠ 해당 함수명 고치고 공통으로 쓰는 category나 event들 setUp으로 빼면 될둡!

jangjichang replied:
아하 그러구나 내가 실수했네 이런걸로 시간 잡아먹게 해서 미안해~ 하하;;

jangjichang committed:
Add Postgresql setting

changed a development database environment from sqlite3 to postgresql

jangjichang committed:
Modify the code according to the code static analysis

jangjichang committed:
Add a feature search by title

Added searching by title.

Added sorting in descending order of the 'start_at' field.
